examples/wallet/app/actions/account.js

```diff
@@ -29,31 +32,94 @@ import {
   getTransactions
 } from '../utils/nodeConnection';
 import { isValidMnemonic, createSeedFromMnemonic } from '../utils/mnemonic';
+import { saveAccountInfoInLocalStorage } from '../utils/storage';
+
 import routes from '../constants/routes.json';
 
 export type SetKeysAction = { type: 'SET_KEYS' } & AccountKeys;
 export const SET_KEYS = 'SET_KEYS';
 
 export function setAccount(privateKey: string): Thunk<SetKeysAction> {
   return function setAccountThunk(dispatch) {
+    console.log('setAccount setAccount setAccount');
     return getAccountFromPrivateKey(privateKey).then(
       curry(initializeKeysAndRedirect)(dispatch)
     );
   };
 }
 
-const initializeKeysAndRedirect = (dispatch, keys: AccountKeys) => {
+export function loadAccountFromPrivateKey(
+  dispatch: Dispatch,
+  privateKey: string
+): void {
+  getAccountFromPrivateKey(privateKey)
+    .then(loadedPrivateKey => {
+      curry(initializeKeysAndRedirect)(dispatch, loadedPrivateKey, false);
```

Comment:
why curry if it's being called here anyway?

Reply:
Dispatched action instead